designer/client/src/components/graph/node-modal/tests/TestValue.tsx

```diff
@@ -11,18 +11,26 @@ function prettyPrint(obj: unknown) {
     return JSON.stringify(obj, null, 2);
 }
 
-export default function TestValue(props: Props) {
+export default forwardRef<HTMLTextAreaElement, Props>(function TestValue(props: Props, ref: React.Ref<HTMLTextAreaElement>) {
     const { value, shouldHideTestResults } = props;
-
     return (
         <div className={cx("node-value", shouldHideTestResults && "partly-hidden")}>
-            {value?.original ? <ReadonlyTextarea value={value.original} /> : null}
-            <ReadonlyTextarea value={prettyPrint(value?.pretty)} />
+            {value?.original ? <ReadonlyTextarea ref={ref} value={value.original} /> : null}
+            <ReadonlyTextarea ref={ref} value={prettyPrint(value?.pretty)} />
             {shouldHideTestResults ? <div className="fadeout" /> : null}
         </div>
     );
-}
+});
 
-function ReadonlyTextarea({ value = "" }: { value: string }) {
-    return <textarea className="node-input" readOnly value={value} rows={value.split("\n").length} />;
-}
+const ReadonlyTextarea = forwardRef<HTMLTextAreaElement, { value: string }>(function ReadonlyTextarea(
+    { value = "" }: { value: string },
+    outerRef: React.Ref<HTMLTextAreaElement>,
+) {
+    const innerRef = useRef<HTMLTextAreaElement>(null);
+    useImperativeHandle(outerRef, () => innerRef.current, []);
+    useEffect(() => {
+        innerRef.current.style.height = innerRef.current.scrollHeight + "px";
+    }, []);
+
```

Comment:
```suggestion
 const textAreaFullHeight = innerRef?.current?.scrollHeight;

    return (
        <textarea
            ref={innerRef}
            style={{ height: textAreaFullHeight }}
            className="node-input"
            readOnly
            value={value}
            rows={value.split("\n").length}
        />
    );
```
Perhaps we could make it clearer and eliminate the need for useEffect?

Reply:
Done, I removed useEffect